classes/local/model/cms.php

```diff
@@ -87,16 +87,35 @@ protected static function define_properties(): array {
      * Includes hashes for the CMS type and the datasources as well, as they
      * contribute to what gets displayed.
      *
+     * @deprecated
      * @return string
+     * @throws \moodle_exception
      */
     public function get_content_hash(): string {
-        $hash = '';
+        throw new \moodle_exception('This method is deprecated. Use get_cache_key instead');
+    }
+
+    /**
+     * Creates and returns a cache key. This key is a hash of the combination of the
+     * datasources, the CMS type and the CMS data.
+     *
+     * @return string|null
+     */
+    public function get_cache_key(): ?string {
+        $key = '';
         foreach (dsbase::get_datasources($this) as $ds) {
```

Comment:
Just a question here, are we certain that `dsbase::get_datasources($this)` always returns values in the same order. I suppose in terms of caching it doesn't really matter too much but order matters when hashing

Reply:
I believe so. The list of plugins is compiled and stored in a special cache, so it ordering is consistent. I looked at the file, and the plugins are listed in alpha order. dsbase::register_datasources uses the plugin list to compile it's list, so it should be consistent.